u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AndroidKeyStoreAttestation.java

```diff
@@ -41,20 +43,23 @@
   private final byte[] attestationChallenge;
   private final AuthorizationList softwareAuthorizationList;
   private final AuthorizationList teeAuthorizationList;
+  private final boolean chainValidated;
+
 
   private AndroidKeyStoreAttestation(Integer keymasterVersion, byte[] attestationChallenge,
-      AuthorizationList softwareAuthorizationList, AuthorizationList teeAuthorizationList) {
+      AuthorizationList softwareAuthorizationList, AuthorizationList teeAuthorizationList,
+      boolean chainValidated) {
     this.keymasterVersion = keymasterVersion;
     this.attestationChallenge = attestationChallenge;
     this.softwareAuthorizationList = softwareAuthorizationList;
     this.teeAuthorizationList = teeAuthorizationList;
+    this.chainValidated = chainValidated;
   }
 
   /**
-   * Parses the key description extension.  Note that this method only parses the description
-   * extension in the leaf cert.  It *does not* validate the certificate (or any chain).
-   *
-   * TODO(aczeskis): Add chain validation and remove/clarify the above comment.
+   * Parses the key description extension out of the leaf node of an AndroidKeyStore attestation
+   * X509 certificate chain.  Note the leaf certificate is considered to be the 0th element in the
+   * certificate array.  The chain is validated up to the root ca certificate;
```

Comment:
Nit: sentences should end in a period.

Another nit: you've introduced a new parameter. @param annotations?